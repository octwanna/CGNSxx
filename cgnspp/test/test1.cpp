/***************************************************************************
                          test1.cpp  -  description
                             -------------------
    begin                : Thu Mar 28 2002
    Copyright (c) 1998-2004 Manuel Kessler, University of Stuttgart
    Copyright (c) 2001-2002 Udo Tremel, EADS Germany

    See the file COPYING in the toplevel directory for details.
 ***************************************************************************/

#include "cgns++.h"

#include "test.h"

#include <iostream>

using std::cerr;
using std::endl;

void CreateTest1(TempFile & file) {
	cerr << "Now we try to create a new CGNS database.\n";
	CGNS::File tempFile(file.mName, CGNS::File::WRITE);
	cerr << "CGNS Open of temporary file (" << file.mName
		<< ") seems to have been successfull.\n";
	CGNS::Base linearBase=tempFile.writeBase("LinearBase", 1, 1);
	int const linearVertices=6;
	cerr << "base has been written\n";
	CGNS::Zone linearStrucZone=linearBase.writeStructured1DZone("LinearStructured", linearVertices);
	CGNS::Zone linearUnstrZone=linearBase.writeUnstructuredZone("LinearUnstructured",
		linearVertices, linearVertices-1, 2);
	cerr << "zones have been written\n";
	CGNS::GridCoordinates linearStrucGrid=linearStrucZone.writeGridCoordinates();
	CGNS::GridCoordinates linearUnstrGrid=linearUnstrZone.writeGridCoordinates();
	cerr << "GridCoordinates have been written\n";
	CGNS::DataArray linearStrucX=linearStrucGrid.writeCoordinate(CGNS::GridCoordinates::COORDINATE_X);
	CGNS::DataArray linearUnstrX=linearUnstrGrid.writeCoordinate(CGNS::GridCoordinates::COORDINATE_X);
	cerr << "coordinates have been created\n";
	std::vector<double> coorX(linearVertices, 0);
	for (unsigned i=0; i<coorX.size(); ++i)
		coorX[i]=i;
	linearStrucX.writeData(&coorX[0]);
	coorX[0]=-linearVertices+1;
	for (unsigned i=1; i<coorX.size(); ++i)
		coorX[i]=1.0-i;
	linearUnstrX.writeData(&coorX[0]);
	cerr << "coordinates have been written\n";
	CGNS::FlowSolution linearStrucSol=linearStrucZone.writeFlowSolution("Solution");
	cerr << "solution has been created\n";
	CGNS::DataArray linearStrucVx=linearStrucSol.writeField(CGNS::FlowSolution::VELOCITY_X);
	std::vector<double> vX(linearVertices, 0);
	for (unsigned i=0; i<vX.size(); ++i)
		vX[i]=i*0.5;
	linearStrucVx.writeData(&vX[0]);
	cerr << "solution has been written\n";
	cerr << "########### Test1: Finished. Ok! ###########\n";
}

