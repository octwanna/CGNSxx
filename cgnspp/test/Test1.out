ADF MotherNode{Root Node of ADF File}:MT
+-CGNSLibraryVersion{CGNSLibraryVersion_t}:R4[1]=2.1
+-LinearBase{CGNSBase_t}:I4[2]={1,1}
  +-LinearStructured{Zone_t}:I4[1,3]
  | +-GridCoordinates{GridCoordinates_t}:MT
  | | +-CoordinateX{DataArray_t}:R8[6]={0,1,2,3,4,5}
  | +-Solution{FlowSolution_t}:MT
  | | +-VelocityX{DataArray_t}:R8[6]={0,0.5,1,1.5,2,2.5}
  | +-ZoneType{ZoneType_t}:C1[10]="Structured"
  +-LinearUnstructured{Zone_t}:I4[1,3]
    +-GridCoordinates{GridCoordinates_t}:MT
    | +-CoordinateX{DataArray_t}:R8[6]={-5,0,-1,-2,-3,-4}
    +-ZoneType{ZoneType_t}:C1[12]="Unstructured"
